### PR TITLE
feat: Bearer + expiration verification + test

### PR DESCRIPTION
**Pull request by vladimirovmm**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -184,15 +187,20 @@ where
...
+        let Some(Ok(auth_str)) = req.headers().get(AUTHORIZATION).map(|auth| auth.to_str()) else {
             let response = Response::builder()
                 .status(StatusCode::UNAUTHORIZED)
                 .body(Default::default())
                 .unwrap();
             return futures::future::Either::Right(futures::future::ok(response));
         };
 
-        if self.jwt.decode(token).is_err() {
+        if auth_str[..Bearer::SCHEME.len()].to_lowercase() != Bearer::SCHEME.to_lowercase()
```
Comment (i1i1) [inline src/lib.rs]: This will panic if auth_string will be `""` and in all cases when `auth_str.len() < Bearer::SCHEME.len()`

File: src/lib.rs [hunk 2]
```diff
@@ -196,11 +196,14 @@ where
             return unauthorized();
         };
 
-        if auth_str[..Bearer::SCHEME.len()].to_lowercase() != Bearer::SCHEME.to_lowercase() {
+        let bearer_len = Bearer::SCHEME.len();
+        if auth_str.len() <= bearer_len
+            || auth_str[..bearer_len].to_lowercase() != Bearer::SCHEME.to_lowercase()
```
Comment (i1i1) [inline src/lib.rs]: This should be better though
```rust
auth_str.to_lowercase().strip_prefix(Bearer::SCHEME.to_lowercase())
```

File: Cargo.toml [hunk 3]
```diff
@@ -16,6 +16,10 @@ tower = "0.4"
 
 [dev-dependencies]
 hyper = "0.14.29"
+jsonrpc-core = "18.0.0"
+jsonrpc-core-client = "18.0.0"
+jsonrpsee = {version = "0.22.5", features = ["server", "http-client", "macros"]}
+jsonrpsee-http-client = "*"
```
Comment (i1i1) [inline Cargo.toml]: You only need `jsonrpsee-http-client` here

File: src/lib.rs [hunk 4]
```diff
@@ -146,7 +146,15 @@ where
...
     fn call(&mut self, mut req: Request<B>) -> Self::Future {
         if let Ok(claim) = self.jwt.claim() {
-            req.headers_mut().insert(AUTHORIZATION, claim);
+            let auth_bytes = format!("{} ", Bearer::SCHEME)
+                .as_bytes()
+                .iter()
+                .chain(claim.as_bytes())
+                .cloned()
+                .collect::<Vec<_>>();
+            let auth_head = HeaderValue::from_bytes(&auth_bytes).expect("An unexpected error.");
```
Comment (i1i1) [inline src/lib.rs]: This should never panic. Lets explain in `expect` why :D

File: src/lib.rs [hunk 5]
```diff
@@ -146,7 +146,15 @@ where
 
     fn call(&mut self, mut req: Request<B>) -> Self::Future {
         if let Ok(claim) = self.jwt.claim() {
-            req.headers_mut().insert(AUTHORIZATION, claim);
+            let auth_bytes = format!("{} ", Bearer::SCHEME)
+                .as_bytes()
+                .iter()
+                .chain(claim.as_bytes())
+                .cloned()
+                .collect::<Vec<_>>();
```
Comment (i1i1) [inline src/lib.rs]: This seems like an overkill. Move it to `JwtSecret::claim` so that header value is constructed there

File: src/lib.rs [hunk 6]
```diff
@@ -327,4 +340,53 @@ mod test {
...
+
+        let mut module = RpcModule::new(());
+        module.register_method("hello", |_, _| "hello").unwrap();
+
+        let _server = jsonrpsee::server::Server::builder()
+            .set_http_middleware(service_builder)
+            .build(ADDRESS)
+            .await
+            .unwrap()
+            .start(module);
```
Comment (i1i1) [inline src/lib.rs]: This part is repeated twice, lets extract it somewhere

File: src/lib.rs (hunk 4 above)
Comment (vladimirovmm) [reply src/lib.rs]: The line has been deleted

File: Cargo.toml (hunk 3 above)
Comment (vladimirovmm) [reply Cargo.toml]: The lines has been deleted

**Pull request merged**